typescript/cli/src/tests/warp/warp-apply.e2e-test.ts

```diff
@@ -802,4 +802,39 @@ describe('hyperlane warp apply e2e tests', async function () {
       ].address.toLowerCase(),
     ).to.equal(newRouterAddress.toLowerCase());
   });
+
+  it('should preserve deploy config when extending warp route', async () => {
+    const warpDeployPath = `${TEMP_PATH}/warp-route-deployment-2.yaml`;
+
+    const warpDeployConfig = await readWarpConfig(
+      CHAIN_NAME_2,
+      WARP_CORE_CONFIG_PATH_2,
+      warpDeployPath,
+    );
+
+    // Extend with new config for chain 3
+    const extendedConfig: HypTokenRouterConfig = {
+      decimals: 18,
+      mailbox: chain2Addresses!.mailbox,
+      name: 'Ether',
+      owner: new Wallet(ANVIL_KEY).address,
+      symbol: 'ETH',
+      totalSupply: 0,
+      type: TokenType.native,
+    };
+
+    warpDeployConfig[CHAIN_NAME_3] = extendedConfig;
+    // Remove remoteRouters and destinationGas as they are written in readWarpConfig
```

Comment:
Could we not just get the warpDeployConfig without reading it? Would much prefer that to simulate the actual usage and therefore test